train.py

```diff
@@ -107,6 +110,11 @@ def save_model(self, key, save_path=None, verbose=False):
     def get_models(self):
         return self.models
 
+    def on_epoch_end(self, epoch):
```

Comment:
you don't always want to save all the models, just the ones that improved, no?

Reply:
This vanilla function saves all the three, we may extend it with more logic